### DISCUSSION TITLE
Add the concept of urgency for a pane. Also a new color category like `config.colors.tab_bar.urgent_tab`

### DISCUSSION THREAD

**Discussion by natrys** [wezterm/wezterm#4064] [Ideas]
A feature I am missing from urxvt is `URxvt.urgentOnBell: true` that sets window urgency on bell.

Firstly, I realise there is a way to trigger toast notification, from bell event handler. But I think the toast notification is fairly obtrusive and doesn't provide actionable interface, like it doesn't give me a visual cue about which inactive tab have become urgent (and in which window). I would like to be able to deduce that by simply looking at the wezterm tab bar.

Secondly, for system-wide notification, I would prefer to have access to API that can set/reset window urgency hint (X11). I would like that over toast notification because:

- Visually less intrusive. Most DEs and independent bars (like polybar) knows to only highlight the relevant workspace, which is all the visual cue that one needs.
- Hotkey support. Since urgency is a distinct concept, the WMs provide a way to switch to workspace containing the urgent window with a single hotkey.

On the flip side, this is probably married to X11, whereas wezterm needs to support way more things. So I would understand the reluctance about the second proposal. Still, I am not asking for full blown notification feature, just the low level API.

**Comment by wez** [comment]
Please consider filing this as an enhancement request issue so that it is searchable when performing issue triage.
You may also wish to look through:
* https://github.com/wez/wezterm/pull/1636
